src/main/java/seedu/address/logic/parser/AddShortcutCommandParser.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMMAND_WORD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SHORTCUT;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.AddShortcutCommand;
+import seedu.address.logic.commands.CommandWord;
+import seedu.address.logic.commands.ShortcutAlias;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new AddShortcutCommand object
+ */
+public class AddShortcutCommandParser implements ParserBasic<AddShortcutCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddShortcutCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AddShortcutCommand parse(String userInput) throws ParseException {
+        ArgumentMultimap argMultimap =
+
+                ArgumentTokenizer.tokenize(userInput, PREFIX_COMMAND_WORD, PREFIX_SHORTCUT);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_SHORTCUT, PREFIX_COMMAND_WORD)
+
+                || !argMultimap.getPreamble().isEmpty()) {
+
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddShortcutCommand.MESSAGE_USAGE));
+        }
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_COMMAND_WORD, PREFIX_SHORTCUT);
+
+        CommandWord commandWord = ParserUtil.parseCommandWord(argMultimap.getValue(PREFIX_COMMAND_WORD).get());
+        ShortcutAlias shortcutAlias = ParserUtil.parseShortcutAlias(argMultimap.getValue(PREFIX_SHORTCUT).get());
+
+        return new AddShortcutCommand(shortcutAlias, commandWord);
+    }
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
This method can be moved into `ArgumentMultimap` class.

Reply:
Agreed, will remove the copy of this method from `AddCommandParser` as well